File: src/test/console/java/src/main/java/web/MultipleWindowsPage.java
package web;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class MultipleWindowsPage extends Base {
  @FindBy(xpath= "//*[@id=\"content\"]/div/h3[text() = \"Opening a new window\"]")
  public WebElement header;
  @FindBy(xpath = "//*[@id=\"content\"]/div/a[contains(.,\"Click Here\")]")
  public WebElement clickHere;

  public MultipleWindowsPage(WebDriver driver, long timeoutInSeconds) {
    super(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/windows");
  }

  public void open() throws Exception{
    super.open();
    this.waitForElementVisible(header);
  }

  public void createWindows(Integer numOfWindows) {
    for (int i = 0; i< numOfWindows; i++) {
      this.clickHere.click();
    }
  }

  public void closeCreatedWindows(String baseWindow){
    for(String handle: this.getDriver().getWindowHandles() ){
      // Switch to window handle
      if(!baseWindow.equals(handle)){
        this.getDriver().switchTo().window(handle);
        this.getDriver().close();
      }
    }
  }

}

File: src/test/console/dotnet/Jsonwired/web/MultipleWindowsPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace web {
	public class MultipleWindowsPage : Base {
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/h3[text() = \"Opening a new window\"]")]
		public IWebElement header ;
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/a[contains(.,\"Click Here\")]")]
		 public IWebElement clickHere ;

		public MultipleWindowsPage(IWebDriver driver, long timeoutInSeconds):
			base(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/windows") {
			}

		public override void open() {
			base.open();
			this.waitForElementVisible(header);
		}

		 public void createWindows(int? numOfWindows) {
			for (int i = 0; i < numOfWindows; i++) {
				this.clickHere.Click();
			}
		}

			public void closeCreatedWindows(string baseWindow) {
				foreach (string handle in this.getDriver().WindowHandles) {
					// Switch to window handle
					if (!baseWindow.Equals(handle)) {
					this.getDriver().SwitchTo().Window(handle);
					this.getDriver().Close();
					}
				}
			}
	}
}
